packages/dapp/src/components/EndorseForm/index.tsx

```diff
@@ -282,8 +258,79 @@ export const EndorseForm = ({ endorsee }: EndorseeProps) => {
   };
 
   useEffect(() => {
-    if (isReceiptLoading) setEndorsementModalOpen(true);
-  }, [isReceiptLoading]);
+    if (isReceiptLoaded) setEndorsementModalOpen(true);
+  }, [isReceiptLoaded]);
+
+  useEffect(() => {
+    if (transactionHash === undefined) return;
+    const transactionReceipt = waitForTransactionReceipt(config, {
```

Comment:
We already use `useWaitForTransactionReceipt` above.

Reply:
This one is needed for its promise so toast.promise works. The other one is used for everything else.